### PR TITLE
Push template and some flags

### PR DESCRIPTION
**Pull request by zealic**
For issue: #111 #205 #206

### CODE REVIEW

File: api/v1/v1.go [hunk 1]
```diff
@@ -52,6 +59,10 @@ type PushConfig struct {
 	UpdateChanged bool
 	// PathSeparator defines which path separator to use (default: "/")
 	PathSeparator string
+	// PathTemplate is a template to change push path, sprig functions are supprted
```
Comment (ivanilves) [inline api/v1/v1.go]: `s/supprted/supported/` 🙏

File: api/v1/v1.go [hunk 2]
```diff
@@ -52,6 +59,10 @@ type PushConfig struct {
...
+	PathTemplate string
+	// TagTemplate is a template to change push tag, sprig functions are supprted
```
Comment (ivanilves) [inline api/v1/v1.go]: `s/supprted/supported/` 🙏

**Comment by zealic** [comment]
> Awesome contribution @zealic 🚀
> 
> A minor orthography comment only.

@ivanilves Thank you for your compliment, but I want to fix the problem of CI coverage failure. Now I don't have time, can you fix this?

**Comment by ivanilves** [comment]
YES! Of course - merging this cause I can not easily change branch on your fork AFAIK.

Then I will create another PR with fix of tests. In this repo having broken master is safe as long as release pipeline will not make a release from the code failing tests and release will be created after merge of test fix PR. 😉

**Pull request merged**